### DISCUSSION TITLE
Converting the Smecno Demo sampleset

### DISCUSSION THREAD

**Discussion by oleg68** [GrandOrgue/OdfEdit#13] [Q&A] [answered]
I'm trying to convert [the Smecno demo sampleset](https://www.sonusparadisi.cz/en/blog/smecno-demo-sample-set/) from HW to GrandOrgue.

There are the following problems:

1. The demo hw definition file has the `.Organ.Hauptwerk.xml` extension instead of the `.Organ_Hauptwerk_xml` expected by OdfEdit.
2. When I'm trying to open the resulting ODF with GrandOrgue, it can not with the error message ``../OrganInstallationPackages/000833/html/index.html' does not exist``. The reason: the file has the `.htm` extension instead of `.html`.
3. After the manual correction, the next error is ``Out of range value at section 'Stop042' entry 'Switch001': 0``

    The reason is the following values in the [Stop042] section:

    ```
    SwitchCount=1
    Switch001=
    ```

    The same error presents in lots of stops, so I can not correct it manually.

**Comment by eturpault** [comment, marked as answer]
Thanks @oleg68 for this interresting new test case ! I answer to your points one by one : 

1. I will add in OdfEdit v2.2, on which I am currently working, the support of HW ODF with .xml extension. If one select a xml which is not a real HW ODF, it will be detected when OdfEdit checks the presence of the tag Hauptwerk FileFormat="Organ" in the xml file. This sample set has an ODF with file format version 3.20, whereas all the one tested so far have the version 4.0 or 4.2, maybe it is the reason why the file extension is not the same.
2. from OdfEdit v2.2 the attribute InfoFilename of the Organ section will no more be present in the GO ODF, as we discussed in another discussion, so this issue will not occur anymore.
3. in OdfEdit v2.2 I reworked a lot the way to build the switches and stops in order to be more robust to the various sample sets. With this version I have no more the issue that you observed with undefined Switch attribute in Stop objects.
I am doing GUI improvements in OdfEdit : adding paned windows to permit to resize manually the width of the objects list and tree and attributes editor. Once it is completed I will release the v2.2, which supports the enclosures and volume sliders.
